runtime error: invalid memory address or nil pointer dereference when assemble-runtime creates image_metadata.json

```shell
s2i build https://github.com/3scale/apicast.git quay.io/3scale/s2i-openresty-centos7:master apicast-runtime-test --context-dir=gateway --runtime-image=quay.io/3scale/s2i-openresty-centos7:master-runtime --pull-policy=always --runtime-pull-policy=always --env GIT_BRANCH=quay-tag-expiration --ref=s2i-assemble-runtime-segfault
```

version 1.1.7 fails with

```output
---> Copying runtime artifacts
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x137e7c6]

goroutine 1 [running]:
github.com/openshift/source-to-image/pkg/build/strategies/sti.downloadAndExtractFileFromContainer(0x166ed20, 0xc420352180, 0x0, 0x0, 0xc4200e50a0, 0x1d, 0xc42030cb40, 0x46, 0xc4201c8140, 0x40, ...)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:494 +0x6a6
github.com/openshift/source-to-image/pkg/build/strategies/sti.checkAndGetNewLabels(0xc4202b6e00, 0x166ed20, 0xc420352180, 0x0, 0x0, 0xc4201c8140, 0x40, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:535 +0x58c
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*commitImageStep).execute(0xc42030d090, 0xc42030d040, 0x100c8e8, 0x140b320)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:124 +0x29c
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).PostExecute(0xc4202b6e00, 0xc4201c8140, 0x40, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:434 +0xe6
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer.func4(0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1066 +0xae8
github.com/openshift/source-to-image/pkg/util/interrupt.(*Handler).Run(0xc4203200c0, 0xc42053c1b0, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/util/interrupt/interrupt.go:78 +0x133
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer(0xc420352180, 0x7ffeefbff647, 0x33, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1071 +0xa73
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*startRuntimeImageAndUploadFilesStep).execute(0xc420303800, 0xc42030d040, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:359 +0xe95
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).PostExecute(0xc4202b6e00, 0xc4202c8200, 0x40, 0x1475f47, 0x4, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:434 +0xe6
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer.func4(0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1066 +0xae8
github.com/openshift/source-to-image/pkg/util/interrupt.(*Handler).Run(0xc42007d2c0, 0xc4205cafb0, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/util/interrupt/interrupt.go:78 +0x133
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer(0xc420352180, 0x7ffeefbff5e0, 0x2b, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1071 +0xa73
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).Execute(0xc4202b6e00, 0x147760a, 0x8, 0x0, 0x0, 0xc420089500, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:685 +0x732
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).Build(0xc4202b6e00, 0xc420089500, 0x0, 0x0, 0x0)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:225 +0x75b
main.newCmdBuild.func1(0xc4200a3440, 0xc4202b2dc0, 0x3, 0xa)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/cmd/s2i/main.go:163 +0xa2b
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200a3440, 0xc4202b2c80, 0xa, 0xa, 0xc4200a3440, 0xc4202b2c80)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:603 +0x22b
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200a2fc0, 0xc420195f50, 0x0, 0xffffffffffffffff)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:689 +0x339
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200a2fc0, 0xc420195f48, 0x1)
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/private/tmp/source-to-image-20170623-25568-rt8pch/_output/local/go/src/github.com/openshift/source-to-image/cmd/s2i/main.go:632 +0x63d
```

~~~I guess that is because [assemble-runtime is a symlink](https://github.com/3scale/apicast/blob/3c3d8054f36bfd73e10e464e9e91577025ac3c4a/gateway/.s2i/bin/assemble-runtime).~~~

I can reproduce the issue even when `assemble-runtime` is normal file:

```shell
s2i build https://github.com/3scale/apicast.git quay.io/3scale/s2i-openresty-centos7:master apicast-runtime-test --context-dir=gateway --runtime-image=quay.io/3scale/s2i-openresty-centos7:master-runtime --pull-policy=always --runtime-pull-policy=always --env GIT_BRANCH=quay-tag-expiration --ref=quay-tag-expiration
```

@bparees 1.1.7, updating to 1.1.8 does not fix the issue.

1.1.8 trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1379786]

goroutine 1 [running]:
github.com/openshift/source-to-image/pkg/build/strategies/sti.downloadAndExtractFileFromContainer(0x16a90a0, 0xc4203ea100, 0x0, 0x0, 0xc42030a4c0, 0x1d, 0xc42001a4b0, 0x46, 0xc420184e00, 0x40, ...)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:495 +0x6a6
github.com/openshift/source-to-image/pkg/build/strategies/sti.checkAndGetNewLabels(0xc4200d8400, 0x16a90a0, 0xc4203ea100, 0x0, 0x0, 0xc420184e00, 0x40, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:536 +0x57c
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*commitImageStep).execute(0xc4203e2230, 0xc4203e21e0, 0x144d6a0, 0xc4200ae800)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:125 +0x29c
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).PostExecute(0xc4200d8400, 0xc420184e00, 0x40, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:433 +0x102
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer.func4(0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1079 +0xad7
github.com/openshift/source-to-image/pkg/util/interrupt.(*Handler).Run(0xc420085800, 0xc420634098, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/util/interrupt/interrupt.go:78 +0x133
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer(0xc4203ea100, 0x7ffeefbff60b, 0x33, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1012 +0xa5f
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*startRuntimeImageAndUploadFilesStep).execute(0xc4203b2ab0, 0xc4203e21e0, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/postexecutorstep.go:360 +0xe7d
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).PostExecute(0xc4200d8400, 0xc420088380, 0x40, 0x147834d, 0x4, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:433 +0x102
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer.func4(0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1079 +0xad7
github.com/openshift/source-to-image/pkg/util/interrupt.(*Handler).Run(0xc420196690, 0xc4201dcfb8, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/util/interrupt/interrupt.go:78 +0x133
github.com/openshift/source-to-image/pkg/docker.(*stiDocker).RunContainer(0xc4203ea100, 0x7ffeefbff5a4, 0x2b, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/docker/docker.go:1012 +0xa5f
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).Execute(0xc4200d8400, 0x1479597, 0x8, 0x0, 0x0, 0xc420073c00, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:687 +0x728
github.com/openshift/source-to-image/pkg/build/strategies/sti.(*STI).Build(0xc4200d8400, 0xc420073c00, 0x0, 0x0, 0x0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/build/strategies/sti/sti.go:224 +0x6fc
github.com/openshift/source-to-image/pkg/cmd/cli/cmd.NewCmdBuild.func1(0xc4200b5680, 0xc4200aa180, 0x3, 0xc)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/pkg/cmd/cli/cmd/build.go:153 +0x9cc
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200b5680, 0xc4200aa0c0, 0xc, 0xc, 0xc4200b5680, 0xc4200aa0c0)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:603 +0x234
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200b5200, 0xc4201ddec8, 0x1392682, 0xc4200b5200)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:689 +0x2fe
github.com/openshift/source-to-image/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200b5200, 0x3, 0xc4200b5200)
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/private/tmp/source-to-image-20171110-28766-17xd4hh/_output/local/go/src/github.com/openshift/source-to-image/cmd/s2i/s2i.go:22 +0x1a8
make: *** [runtime-image] Error 2
```

--- a/pkg/build/strategies/sti/sti.go
+++ b/pkg/build/strategies/sti/sti.go
@@ -741,6 +741,7 @@ func (builder *STI) initPostExecutorSteps() {
 				image:   builder.config.RuntimeImage,
 				builder: builder,
 				docker:  builder.docker,
+				tar:     builder.tar,
 			},
 			&reportSuccessStep{
 				builder: builder,

